# Microarray Lab

Current arrays used (including Affymetrix(r) and Agilent(r)) tend to have no 
problems with finding and calculating spot intensities. Therefore, this lab will 
concentrate on exploring data from public databases and performing a basic analysis
using open-source tools such as [`Bioconductor`](http://bioconductor.org).

## Exploration

## Websites

Currently DNA microarray data is deposited in 
 * [GEO](http://www.ncbi.nlm.nih.gov/geo/) 
 * [ArrayExpress](http://www.ebi.ac.uk/arrayexpress/)
 
These both provide a rich source of raw data for analysis. You should spend some time looking at the websites. 

Can we answer some basic questions about the number of data sets in each database?

* How many platforms?
* How many experiments (series)?
* How many samples?
* How many curated datasets?
* How many organisms represented?

How hard is this information to get via the web-site?

### Series

Use the [browse link](www.ncbi.nlm.nih.gov/geo/browse/) to look at the various data sets in 
many different ways. How hard is it to find a diabetes `Series` study that uses human samples, that has CEL files available?

### GPL

The official `GPL` for the Affymetrix(r) human U133 Plus 2.0 genome chip is [`GPL570`](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL570).
However, there are a lot **alternative** platforms. Can you find them all browsing for 
`Platforms`? 

### Datasets

`Datasets` are curated experiments at GEO, that allow investigation of gene expression levels
as well as perform basic analyses using on-line tools. We will use the online tools available at GEO to examine [GDS3324](http://www.ncbi.nlm.nih.gov/sites/GDSbrowser?acc=GDS3324). 

What genes come up as different based on tissue and condition? Hint, there is a link on the 
analysis page.

Pick an analysis to perform. How many genes come up as differentially expressed? Are any of these the control probesets on the chip?
 

### Bioconductor

[`Bioconductor`](http://bioconductor.org) is a set of `R` libraries specifically
developed for high-throughput biological analysis
 * Originally geared towards DNA microarray
 * Now moving more into other -omics and RNASeq
 
You should have downloaded the **CEL** data for the **GSE** corresponding to [GDS3324](http://www.ncbi.nlm.nih.gov/geosuppl/?acc=GSE10797). Use `tar` or [`7zip`](http://www.7-zip.org) to extract
the files to a directory.

Our goal is to use a standard Bioconductor workflow to process the data and get a list of genes
out. Possibly in another lab we will examine the genes for enrichment.

### GEOmetadb

All of the meta-information that was searchable on the GEO website, is also contained in the 
GEO metadb sqlite database. We can explore it offline in Bioconductor using SQL.

```{r geometadb}
require(GEOmetadb)
con <- dbConnect("SQLite", "GEOmetadb.sqlite")
dbListTables(con)

dbListFields(con, "gsm")
```

#### Read Data

Start up `RStudio` and navigate to a directory above the GSE raw files.

```{r housekeeping}
# set default compression on saves, and don't use *Factors* in data frames
options(stringsAsFactors=FALSE)
require(affy)
require(limma)
require(arrayQualityMetrics)

```

```{r readData, eval=FALSE}
# not evaluated because it takes a long time on my 2GB Ram machine
celDir <- "GSE10797_RAW"
setwd(celDir)
celDat <- read.affybatch(filenames=dir())

celDat
setwd("..")
```
```{r generatePhenoDat, eval=FALSE}
con <- dbConnect("SQLite", "GEOmetadb.sqlite")
getDat <- dbGetQuery(con, "select title, gsm, series_id from gsm where series_id in ('GSE10797')")
splitSample <- strsplit(getDat$title, "_", fixed=T)
head(splitSample)
getDat$cancerType <- sapply(splitSample, function(x){x[1]})
head(getDat)
getDat$tissueType <- sapply(splitSample, function(x){x[2]})
getDat$replicate <- sapply(splitSample, function(x){x[3]})
head(getDat)
tmpDat <- pData(celDat)
class(tmpDat)
getDat <- getDat[order(getDat$gsm),]
tail(getDat)
head(getDat)
head(tmpDat)
tmpDat <- cbind(tmpDat, getDat)
pData(celDat) <- tmpDat

.sessionInfo <- sessionInfo()
.timedate <- Sys.time()
save(celDat, .sessionInfo, .timedate, file="celData.RData")
```

```{r sumRMA}
load("celData.RData")
rmaDat <- rma(celDat)
```

```{r qualityMetrics, eval=FALSE}
# this also takes a long time, the files are in the git repo.
arrayQualityMetrics(celDat, outdir="rawData", do.logtransform=T, spatial=F, intgroup=c("tissueType", "cancerType"), force=T)
arrayQualityMetrics(rmaDat, outdir="rmaData", do.logtransform=F, intgroup=c("tissueType", "cancerType"), force=T)
```



