# Microarray Lab

Current arrays used (including Affymetrix(r) and Agilent(r)) tend to have no 
problems with finding and calculating spot intensities. Therefore, this lab will 
concentrate on exploring data from public databases and performing a basic analysis
using open-source tools such as [`Bioconductor`](http://bioconductor.org).

## Exploration

## Websites

Currently DNA microarray data is deposited in 
 * [GEO](http://www.ncbi.nlm.nih.gov/geo/) 
 * [ArrayExpress](http://www.ebi.ac.uk/arrayexpress/)
 
These both provide a rich source of raw data for analysis. You should spend some time looking at the websites. 

Can we answer some basic questions about the number of data sets in each database?

* How many platforms?
* How many experiments (series)?
* How many samples?
* How many curated datasets?
* How many organisms represented?

How hard is this information to get via the web-site?

```{r answer1, echo=FALSE}
# Number of samples, etc: front page of GEO or ArrayExpress. 
# ArrayExpress uses slightly different definitions than GEO, 
# but seems to mostly be the same, with more data.
```

### Series

Use the [browse link](www.ncbi.nlm.nih.gov/geo/browse/) to look at the various data sets in 
many different ways. How hard is it to find a diabetes `Series` study that uses human samples, that has CEL files available?

```{r answer2, echo=F}
# Diabetes samples: http://www.ncbi.nlm.nih.gov/geo/browse/?view=series&search=diabetes&tax=9606&zsort=date&display=20
```

### GPL

The official `GPL` for the Affymetrix(r) human U133 Plus 2.0 genome chip is [`GPL570`](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL570).
However, there are a lot **alternative** platforms. Can you find them all browsing for 
`Platforms`? 

```{r answer3, echo=F}
# Affy U133 Plus 2.0: http://www.ncbi.nlm.nih.gov/geo/browse/?view=platform&display=50&tax=9606&search=U133%20plus%202.0&zsort=date
```

### Datasets

`Datasets` are curated experiments at GEO, that allow investigation of gene expression levels
as well as perform basic analyses using on-line tools. We will use the online tools available at GEO to examine [GDS3324](http://www.ncbi.nlm.nih.gov/sites/GDSbrowser?acc=GDS3324). 

What genes come up as different based on tissue and condition? Hint, there is a link on the 
analysis page.

Pick an analysis to perform. How many genes come up as differentially expressed? Are any of these the control probesets on the chip?
 

### Bioconductor

[`Bioconductor`](http://bioconductor.org) is a set of `R` libraries specifically
developed for high-throughput biological analysis
 * Originally geared towards DNA microarray
 * Now moving more into other -omics and RNASeq
 
You should have downloaded the **CEL** data for the **GSE** corresponding to [GDS3324](http://www.ncbi.nlm.nih.gov/geosuppl/?acc=GSE10797). Use `tar` or [`7zip`](http://www.7-zip.org) to extract
the files to a directory.

Our goal is to use a standard Bioconductor workflow to process the data and get a list of genes
out. Possibly in another lab we will examine the genes for enrichment.

### GEOmetadb

All of the meta-information that was searchable on the GEO website, is also contained in the 
GEO metadb sqlite database. We can explore it offline in Bioconductor using SQL.

```{r geometadb}
require(GEOmetadb)
con <- dbConnect("SQLite", "GEOmetadb.sqlite")
dbListTables(con)

dbListFields(con, "gsm")
```

We can do similar sorts of queries using the SQLite database as we did on the website.

However, some of the information is not linked the same way in GEOmetadb. For example, it is
easy to find **datasets** related to human diabetes, but more complicated to find **series**, 
because the organism is *not* part of the series table. However, we can exploit the fact that 
**GSE** is included in the **GSM** table.

```{r geoQueries}
# Can easily find datasets for human diabetes
sQ <- "SELECT ID, gds, description, type, sample_organism FROM gds WHERE description LIKE '%diabetes%' AND sample_organism='Homo sapiens'"
dbRes <- dbGetQuery(con, sQ)
dbRes

sQ_gse <- "SELECT gse FROM gse WHERE title LIKE '%diabetes%'"
db_gse <- dbGetQuery(con, sQ_gse)
head(db_gse)

gseSrch <- paste("'", db_gse$gse, "'", sep="")
gseSrch <- paste(gseSrch, collapse=",", sep="")
gseSrch <- paste("(", gseSrch, ")", sep="", collapse="")
sQ_gsm <- paste("SELECT * FROM gsm WHERE series_id IN ", gseSrch, 
                " AND organism_ch1 IN ('Homo sapiens')", sep="", collapse="")
db_gsm <- dbGetQuery(con, sQ_gsm)
head(db_gsm)
dim(db_gsm)
length(unique(db_gsm$series_id))
```

Now what about all of the series that are attached to Affymetrix HGU 133 plus 2.0 arrays?

```{r geoQ2}
gplQ <- "SELECT gpl FROM gpl WHERE title LIKE '%U133 plus 2.0%'"
gplRes <- dbGetQuery(con, gplQ)
```

Disconnect database when done.

```{r disconnect}
dbDisconnect(con)
```

### Differential Expression Analysis

Using the data downloaded previously and `Bioconductor` functions, we will attempt to determine
those genes that are differentially expressed. 

```{r housekeeping}
# set default compression on saves, and don't use *Factors* in data frames
options(stringsAsFactors=FALSE)
require(affy)
require(genefilter)
require(limma)
require(arrayQualityMetrics)
require(ggplot2)
require(hgu133a2.db)
```

#### Read Data

```{r readData, eval=FALSE}
# not evaluated because it takes a long time on my 2GB Ram machine
celDir <- "GSE10797_RAW"
setwd(celDir)
celDat <- read.affybatch(filenames=dir())

celDat
setwd("..")
```

Unfortunately, this method makes it difficult to know which samples belong to which class
(cancer, normal, stroma, epithelial, etc). We can use the `GEOmetadb` to get this information, 
however. This is required to enable proper analysis of the data.

#### Sample Info

```{r generatePhenoDat, eval=FALSE}
con <- dbConnect("SQLite", "GEOmetadb.sqlite")
getDat <- dbGetQuery(con, "select title, gsm, series_id from gsm where series_id in ('GSE10797')")
splitSample <- strsplit(getDat$title, "_", fixed=T)
head(splitSample)
getDat$cancerType <- sapply(splitSample, function(x){x[1]})
head(getDat)
getDat$tissueType <- sapply(splitSample, function(x){x[2]})
getDat$replicate <- sapply(splitSample, function(x){x[3]})
head(getDat)
tmpDat <- pData(celDat)
class(tmpDat)
getDat <- getDat[order(getDat$gsm),]
tail(getDat)
head(getDat)
head(tmpDat)
tmpDat <- cbind(tmpDat, getDat)
pData(celDat) <- tmpDat

.sessionInfo <- sessionInfo()
.timedate <- Sys.time()
save(celDat, .sessionInfo, .timedate, file="celData.RData")
rm(list=ls())
```

#### RMA Summarize

`RMA` is one of the more popular methods for summarizing Affymetrix(r) GeneChip(r) data. 
It only uses the **PM** probes, models probe behavior to do background correction, quantile
normalization to force identical signal distributions, and then a median-polish to average
the probe signals and generate a final value.

```{r sumRMA}
load("celData.RData")
rmaDat <- rma(celDat)
.sessionInfo <- sessionInfo()
.timedate <- Sys.time()
save(.sessionInfo, .timedate, rmaDat, file="rmaData.RData")
```

#### Quality Control

```{r qualityMetrics, eval=FALSE}
# this also takes a long time, the files are in the git repo.
arrayQualityMetrics(celDat, outdir="rawData", do.logtransform=T, spatial=F, intgroup=c("tissueType", "cancerType"), force=T)
arrayQualityMetrics(rmaDat, outdir="rmaData", do.logtransform=F, intgroup=c("tissueType", "cancerType"), force=T)
```

You should explore the ArrayQualityMetrics reports ([celDat](rawData/index.html), [rmaDat](rmaData/index.html) ) to see what types of things get reported, and how some discrepancies
are resolved due to normalization. 

```{r cleanUP}
rm(celDat)
```

#### ProbeSet Filtering

There are two opinions on filtering before diff. expression analysis. 
 
 1. Filter out the controls
 2. Filter out everything that may be irrelevent, including probesets with low variance and those that don't map to known genes
 
Due to personal experience, and known issues with skewing results due to changing the numbers
for multiple testing adjustment, I prefer **1**, filtering out as few items as required.

```{r filtering}
rmaFilt <- featureFilter(rmaDat, require.entrez=FALSE, remove.dupEntrez=FALSE,
                         feature.exclude="^AFFX")
rmaInt <- exprs(rmaFilt) # get the actual probeset intensities
```

#### Design Setup

We want to be able to compare between the cancer and normal of each cell type, much like
the original [paper](fulltext.pdf). So we will generate a new variable that is a combination
of disease status and tissue, and use this in our design matrix.

```{r design}
pD <- pData(rmaDat)
pD$cancer.tissue <- paste(pD$cancerType, pD$tissueType, sep=".")

f <- factor(pD$cancer.tissue, levels=unique(pD$cancer.tissue))
design <- model.matrix(~0 + f)
colnames(design) <- c("CE", "CS", "NE", "NS")

rm(rmaDat, rmaFilt)
```

#### Limma

We are using the `limma` package for the diff. expression analysis. `limma` allows one to 
use bayesian methods to *borrow* variance information between genes / probesets. When the 
number of samples is large (say 20 or more), the variance estimates don't change between
borrowing and not. However, when the number of samples is small (say 3 - 5 in each class),
this can be much more important.

`limma` has become a de-facto standard in the analysis of DNA microarrays, and that is
partly why we use it here. We could also do a standard t-test, permuted t-tests, and lots
of other things that have appeard in the literature.

```{r limmaCalc}
fit <- lmFit(rmaInt, design)
contrast.matrix <- makeContrasts("CE - NE", "CS - NS", levels=design)

fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)
```

The package provides some very nice functions to summarize the results. `decideTests` is one
that provides a summary of the number of significant probesets / genes in each contrast.

```{r decideTests}
testRes <- decideTests(fit2)
sum(testRes[,'CE - NE'] == 1)
sum(testRes[,'CE - NE'] == -1)
sum(testRes[,'CS - NS'] == -1)
sum(testRes[,'CS - NS'] == 1)
```

However, `topTable` is probably the normal way you would get access to all the data you 
would want to investigate the results. In this example, we will get all of the results so
we can plot some things.

```{r topTable}
ce.ne <- topTable(fit2, coef='CE - NE', number=Inf, adjust.method="BH", p.value=1)
dim(ce.ne)
head(ce.ne)
```

Lets look at the relationship between fold-change and p-value.

```{r plotDat}
plotDat <- data.frame(lfc=ce.ne$logFC, pval=-1*log10(ce.ne$P.Value))
ggplot(plotDat, aes(x=lfc, y=pval)) + geom_point()

plotDat2 <- data.frame(lfc=ce.ne$logFC, pval=-1*log10(ce.ne$adj.P.Val))
ggplot(plotDat2, aes(x=lfc, y=pval)) + geom_point()
```

#### Gene IDs

Can we figure out which genes are diff. expressed? 

```{r diffGenes}
ce.ne.Diff <- ce.ne[(ce.ne$adj.P.Val <= 0.05),]
ce.ne.Diff$Entrez <- unlist(mget(ce.ne.Diff$ID, hgu133a2ENTREZID, ifnotfound=NA))
```

See some `NA` values. i.e. there is no corresponding Entrez Gene. But Affymetrix(r) doesn't
tend to have probesets for no reason. What else could it be?

```{r diffOther}
ce.ne.Diff$ACC <- unlist(mget(ce.ne.Diff$ID, hgu133a2ACCNUM, ifnotfound=NA))
head(ce.ne.Diff)
```

Ok, so they definitely come from somewhere. Those are really big p-values (even adjusted),
what does the raw data look like? i.e. just how different are they on the probe-scale?

```{r probeLevel}
rm(fit, fit2, rmaInt, plotDat, plotDat2)
load("celData.RData")

pmInt <- pm(celDat)
useProbe <- indexProbes(celDat, which="pm", genenames="823_at")

useInt <- log2(pmInt[as.character(useProbe[[1]]),])
rm(pmInt)

ceIndx <- as.logical(design[,'CE'])
neIndx <- as.logical(design[,'NE'])

ceInt <- useInt[,ceIndx]
neInt <- useInt[,neIndx]
```